modules/deployment/notification.tf

```diff
@@ -0,0 +1,43 @@
+data "aws_iam_policy_document" "sns_codestar_policy" {
+  count = var.enabled && var.codestar_notifications_target_arn == "" ? 1 : 0
+
+  statement {
+    actions = ["sns:Publish"]
+
+    principals {
+      type        = "Service"
+      identifiers = ["codestar-notifications.amazonaws.com"]
+    }
+
+    resources = [aws_sns_topic.notifications[count.index].arn]
+  }
+}
+
+resource "aws_codestarnotifications_notification_rule" "notification" {
+  count = var.enabled ? 1 : 0
+
+  detail_type    = var.codestar_notifications_detail_type
+  event_type_ids = var.codestar_notifications_event_type_ids
+  name           = "${var.service_name}-notifications-${data.aws_region.current.name}"
+  resource       = aws_codepipeline.codepipeline[count.index].arn
+  tags           = var.tags
+
+  target {
+    address = var.codestar_notifications_target_arn == "" ? aws_sns_topic.notifications[count.index].arn : var.codestar_notifications_target_arn
```

Comment:
könnte man das auch eleganter via `coalesce` ausdrücken?

Reply:
seems like this doesn't play well with `count = 0`:

```
on ../../terraform-aws-ecs-fargate/modules/deployment/notification.tf line 26, in resource "aws_codestarnotifications_notification_rule" "notification":
  26:     address = coalesce(var.codestar_notifications_target_arn, aws_sns_topic.notifications[count.index].arn)
    |----------------
    | aws_sns_topic.notifications is empty tuple
    | count.index is 0

The given key does not identify an element in this collection value.
```

Reply:
Danke fürs Versuchen. flat_map! that shit